### ISSUE TITLE
NMS time limit 10.0s exceeded

### ISSUE THREAD

**Issue by tak-s** [XinzeLee/PolygonObjectDetection#2]
Thanks for sharing great works!

I am trying to train coco dataset.
When calculating mAP for val data, I got bellow warning.

`WARNING: NMS time limit 10.0s exceeded`

I think when found many bbox, nms cost is too much. then time limit exceeded.

So, I set/change conf_thres=0.1(default is 0.001), it's work no warning.

But, I am afraid that this change will affect learning performance. What do you think?

**Comment by XinzeLee** [comment]
It is correct that if increasing the conf_thres to a higher value (like 0.01) can work without warning. But modify the conf_thres from 0.001 to above may affect the "actual" evaluation of the final performance.
By the way, it is quite normal for the warning in the early stages of the training, since the network is still learning the characteristics of the classes. As a result, in early-stage, the network "almost-randomly" gives a lot of guesses, since it is not sure about the correct characteristics! After some training epochs (maybe several tens or hundreds, depending on dataset), the network will be more certain about its predictions, and the warning will disappear naturally.

**Comment by tak-s** [comment]
Thank you.
I understood it well.

By the way, do you train with the COCO dataset?
I trained 300 epochs. I got mAP@.5=0.31 and mAP_0.5:0.95=0.16 with yolov5-s. Is this are reasonable?
![image](https://user-images.githubusercontent.com/20725737/129298490-f171a610-0b10-4c7d-8496-1e91ed959b5c.png)

**Comment by XinzeLee** [comment]
I haven't tried the full coco dataset. It is really good that you did the experiments! 

I have tried with a small portion of classes of coco, such as beds, couch, cups, laptop, monitor, remotes, etc. From my perspective, the performance of mAP@.5=0.31 and mAP_0.5:0.95=0.16 are already pretty good considering that polygon labels are much harder than bbox (cx, cy, width, height). Intuitively, to get good performance for polygon labels, we need more data than bbox labels. 

Can I check with you how did you get the polygon labels for coco dataset? Did you use the Polygon-Tutorial 2 to convert the segmentation labels to polygon labels? If that so, there might be a intrinsic vital problem with the labels: some segmentations are disjointed. These kinds of labels might cause trouble for the model to learn.
![捕获](https://user-images.githubusercontent.com/87064748/129299645-03fda273-81c0-4397-ba59-dd2b8c6141fc.JPG)

**Comment by tak-s** [comment]
Oh, that's right.
Now my annotation for this repository is separated...

![image](https://user-images.githubusercontent.com/20725737/129306825-a798d78a-daf7-4640-a4ba-192e9d73cba7.png)

This annotation are two bag, two people. But, the correct polygon annotation is one bag and one person.

I will try to fix it.

**Comment by XinzeLee** [comment]
Yes, that is the case. If you found something, please let me know. Thanks!

**Comment by tak-s** [comment]
Hi.
At seg2poly()

```
                    # if not crowded, use segmentation
                    if not object0['iscrowd']:
                        segments = [oneD2twoD(segment+segment[:2]) 
                                           for segment in object0['segmentation']]

```
change to below.

```
                    # if not crowded, use segmentation
                    if not object0['iscrowd']:
                        segments = []
                        for segment in object0['segmentation']:
                            segments.extend(oneD2twoD(segment+segment[:2]))
                        segments = [segments]
```

Since the separated information is connected, a connecting line is drawn when plotting, but I think that there is no problem in finding the quadrangle with minimum_rotated_rectangle.

![image](https://user-images.githubusercontent.com/20725737/129315684-c905cb90-d73a-434a-9463-61ba6347da46.png)

**Comment by XinzeLee** [comment]
Really nice work! The disjointed problems for some connective objects are solved. But I have observed another issue: for class like "skis" as follows, both of the skis on left foot and right foot are connected together. I further looked into the coco dataset, and found that "pair of skis" is considered as a whole. Is that true?
![图片](https://user-images.githubusercontent.com/87064748/129323273-539dc4f2-fae5-4391-b715-b7390b547c9e.png)


Thanks for your help! I will definitely mention your contribution in the new version of codes! Also, if it is possible, can you train the polygon-yolov5s on the new version of labels and test the performance?

**Comment by tak-s** [comment]
Thank you for checking my code!
I'm glad to contribute this work.

My environment is poor and may take a few days, but I'll try to train with the modified labels.

**Comment by tak-s** [comment]
On yolo-l, 
mAP@.5: = 0.424
mAP@.5:.95= 0.238
